examples/multimodal_agent.py

```diff
@@ -49,6 +49,10 @@ async def get_weather(
 
     await ctx.connect(auto_subscribe=AutoSubscribe.AUDIO_ONLY)
 
+    print("waiting for participant")
+    await ctx.wait_for_participant()
```

Comment:
```suggestion
    participant = await ctx.wait_for_participant()
```

Reply:
Do you think we have to wait for both, the `wait_for_subscription` + `wait_for_participant`?
How is that handled in the server side. Is it possible for `wait_for_subscription` to returns before `wait_for_participant`

Reply:
I think we should wait for both.
* wait_for_participant - ensures that we know who we are talking to, so we can set up the agent correctly
* wait_for_subscription - ensures the participant can listen to the agent, so we aren't talking into the void